components/algae_manipulator.py

```diff
@@ -129,5 +130,8 @@ def execute(self) -> None:
             self.flywheel_1.set_control(VelocityVoltage(self.desired_flywheel_speed))
             self.flywheel_2.set_control(VelocityVoltage(self.desired_flywheel_speed))
 
+        if self.desired_injector_speed == 0.0 and not self.algae_limit_switch.get():
```

Comment:
if this is checking for if we have an algae, we already have a method for checking that
```suggestion
        if self.desired_injector_speed == 0.0 and self.has_algae():
```